lib/backend.cc

```diff
@@ -1111,29 +1117,59 @@ void ClDevice::setupProperties(int index) {
 
   // totally made up
   Properties.regsPerBlock = 64;
-  // intel GPU uses minimum 8, so for now set this to 8.
-  // TODO should be properly detected and/or workarounds for __shfl should exist.
-  Properties.warpSize = 8;
+
+  // The maximum subgroup size on an intel GPU
+  if (Dev.getInfo<CL_DEVICE_TYPE>() == CL_DEVICE_TYPE_GPU) {
+    std::vector<uint> sg = Dev.getInfo<CL_DEVICE_SUB_GROUP_SIZES_INTEL>();
+    Properties.warpSize = *std::max_element(sg.begin(), sg.end());
+  }
   Properties.maxGridSize[0] = Properties.maxGridSize[1] =
       Properties.maxGridSize[2] = 65536;
   Properties.memoryClockRate = 1000;
   Properties.memoryBusWidth = 256;
-  Properties.major = 2;
-  Properties.minor = 0;
+
+  Temp = Dev.getInfo<CL_DEVICE_VERSION>();
+  size_t dig_pos = Temp.find_first_of("123456789");
+  size_t dot_pos = Temp.find_first_of(".", dig_pos);
+  size_t spa_pos = Temp.find_first_of(" ", dot_pos);
+  Properties.major = stoi(Temp.substr(dig_pos, dot_pos-1));
+  Properties.minor = stoi(Temp.substr(dot_pos+1, spa_pos-1));
+
   Properties.maxThreadsPerMultiProcessor = 10;
 
   Properties.computeMode = 0;
   Properties.arch = {};
-  Properties.arch.hasGlobalInt32Atomics = 1;
-  Properties.arch.hasSharedInt32Atomics = 1;
-  Properties.arch.hasGlobalInt64Atomics = 1;
-  Properties.arch.hasSharedInt64Atomics = 1;
+
+  Temp = Dev.getInfo<CL_DEVICE_EXTENSIONS>();
+  if (Temp.find("cl_khr_global_int32_base_atomics") != std::string::npos)
+    Properties.arch.hasGlobalInt32Atomics = 1;
+  else
+    Properties.arch.hasGlobalInt32Atomics = 0;
+
+  if (Temp.find("cl_khr_local_int32_base_atomics") != std::string::npos)
+    Properties.arch.hasSharedInt32Atomics = 1;
+  else
+    Properties.arch.hasSharedInt32Atomics = 0;
+
+  if (Temp.find("cl_khr_int64_base_atomics") != std::string::npos) {
+    Properties.arch.hasGlobalInt64Atomics = 1;
+    Properties.arch.hasSharedInt64Atomics = 1;
+  }
+  else {
+    Properties.arch.hasGlobalInt64Atomics = 1;
+    Properties.arch.hasSharedInt64Atomics = 1;
+  }
+
+  if (Temp.find("cl_khr_fp64") != std::string::npos) 
+    Properties.arch.hasDoubles = 1;
+  else
+    Properties.arch.hasDoubles = 0;
 
   Properties.clockInstructionRate = 2465;
   Properties.concurrentKernels = 1;
   Properties.pciDomainID = 0;
   Properties.pciBusID = 0x10;
-  Properties.pciDeviceID = 0x40 + index;
+  Properties.pciDeviceID = Dev.getInfo<CL_DEVICE_VENDOR_ID>();
```

Comment:
Unfortunately this won't work. `CL_DEVICE_VENDOR_ID` is the Vendor ID, not the Device ID.

The PCI Vendor ID for Intel is always `0x8086`; this would make both `pciBusID` and `pciDeviceID` constant for all devices found by HipCL. The combination should be unique for every device. CUDA documentation for `cudaGetDeviceProperties` says:

``pciDeviceID is the PCI device (sometimes called slot) identifier of the device.``

Both AMD and Nvidia have OpenCL extensions to get PCI IDs, but i couldn't find anything for Intel (perhaps you could ask your Intel contacts).